### PR TITLE
ESP32 SPI

### PR DESCRIPTION
**Pull request by pelikhan**
Support for talking to ESP32 via SPI.

### CODE REVIEW

**Comment by sabas1080** [comment]
this interests me a lot, work? Is WIFInina ? is compatible with Arduino MKR1010 and  my project https://github.com/ElectronicCats/escornabot 😍

**Comment by pelikhan** [comment]
This will drive a ESP32 via SPI pins.

**Comment by sabas1080** [comment]
Work in SAMD21 or in SAMD51?

**Comment by pelikhan** [comment]
This will work on all MCUs since it is a SPI protocol.

**Comment by sabas1080** [comment]
Any update on this?

**Comment by sabas1080** [comment]
Escornabot Makech and Meow Meow have an ESP32 expansion https://github.com/ElectronicCats/Backpacks_Meow https://github.com/ElectronicCats/escornabot

File: libs/esp32spi/adafruit_esp32spi.ts [hunk 1]
```diff
@@ -0,0 +1,704 @@
...
+        // size byte
+        // 2 of em here!
+        // we may need more space
+        // handle parameters here
+        // %d is %d bytes long" % (i, len(param)))
+        // wait up to 1000ms
+        // ok ready to send!
+        // pylint: disable=no-member
+        // pylint: disable=too-many-branches
+        private _read_byte(): number {
```
Comment (pelikhan) [inline libs/esp32spi/adafruit_esp32spi.ts]: this one is dubious

File: libs/esp32spi/adafruit_esp32spi.ts [hunk 2]
```diff
@@ -0,0 +1,704 @@
...
+            return this._pbuf[0]
+        }
+
+        /*
+        private _read_bytes(spi: SPI, buffer: Buffer, start: number = 0, end: number = null): void {
+            if (!end) {
+                end = buffer.length
+            }
+
+            spi.transfer(undefined, buffer); // TODO
```
Comment (pelikhan) [inline libs/esp32spi/adafruit_esp32spi.ts]: we don't support range for reading into a buffer.

Comment (mmoskal) [reply libs/esp32spi/adafruit_esp32spi.ts]: this doesn't seem to be used

File: libs/esp32spi/adafruit_esp32spi.ts (hunk 1 above)
Comment (mmoskal) [reply libs/esp32spi/adafruit_esp32spi.ts]: `return this._spi.write(0)`

**Comment by pelikhan** [comment]
@sabas1080 we support the nina-fw protocol (used in Adafruit ESP32 boards) over SPI.

**Comment by sabas1080** [comment]
Nice, this week I'm going to try in escornabot

**Comment by pelikhan** [comment]
@jwunderl support for long lines in console.log

**Pull request merged**